CIP-0000.md

```diff
@@ -0,0 +1,198 @@
+
+Because the CIPs are maintained as text files in a versioned repository, their revision history is the historical record of the feature proposal.
+
+Participation in the Celo project and the CIP process is informed by the Celo [Community Tenets](https://celo.org/community) and subject to the  [Code of Conduct](https://celo.org/code-of-conduct).
+
+# CIP Types
+There are three types of CIP:
+
+* A **Standards Track CIP** describes any change that affects most or all Celo implementations, such as a change to the network protocol, a change in block or transaction validity rules, proposed application standards/conventions, or any change or addition that affects the interoperability of applications using Celo. Furthermore Standard CIPs can be broken down into the following categories. Standards Track CIPs consist of two parts, a design document and implementation.
+	* **Core** - improvements requiring a consensus fork, as well as changes that are not necessarily consensus critical but may be relevant to  “core dev” discussions, and the validator/node strategy.
+	* **Governance** - improvements to the governance proposal to the Celo platform including parameter changes. It should explain the reason for the change, enumerate the smart contract methods that will be called with parameters, and describe the impact and any risks. 
+	* **Networking** - improvements to networking protocol specifications such as devp2p, and proposed improvements like whisper and swarm.
+	* **Interface** - improvements around client [API/RPC] specifications and standards, and also certain language-level standards like method names and contract ABIs.
+	* **CRC** - (Celo [Request for Comments](https://en.wikipedia.org/wiki/Request_for_Comments)) application-level standards and conventions, including contract standards such as token standards, name registries, URI schemes, library/package formats, and wallet formats.
+* A **Meta CIP** describes a process surrounding Celo or proposes a change to (or an event in) a process. Process CIPs are like Standards Track CIPs but apply to areas other than the Celo protocol itself. They may propose an implementation, but not to Celo’s codebase; they often require community consensus; unlike Informational CIPs, they are more than recommendations, and users are typically not free to ignore them. Examples include code of conduct updates, procedures, guidelines, changes to the decision-making process, and changes to the tools or environment used in Celo development. Any Meta CIP is also considered a Process CIP.
+* An **Informational CIP** describes a Celo design issue, or provides general guidelines or information to the Celo community, but does not propose a new feature. Informational CIPs do not necessarily represent Celo community consensus or a recommendation, so users and implementers are free to ignore Informational CIPs or follow their advice.
```

Comment:
I don't quite understand why this exists as part of the CIP process. IIUC, this might be something like a description of how a node operator might execute a key rotation, or some recommendations to users on how to pick a validator for voting. If that it correct, then the forum or documentation website (docs.celo.org) seem like better places for these.

Reply:
I think that certain standards ought to be documented alongside the protocol. An example of an informational BIP that should certainly live in the BIPs repo is the BIP32/39/44/49/84 suite. They describe key generation, serialization, and portability formats that are widely used throughout the crypto ecosystem. While you can use Bitcoin without them, your app will not be able to participate in the ecosystem fully without them

Reply:
Very interesting. What do you think would be a good heuristic for distinguishing such a informational CIP to a Ring 2 or 3 CIP. Arguably, key derivation and portability are pretty crucial for wallet interoperability at this point, however that certainly wasn't the case when these BIPs were originally authored (I assume). Could they "graduate" in that sense?

Reply:
Very interesting. What do you think would be a good heuristic for distinguishing such a informational CIP to a Ring 2 or 3 CIP. Arguably, key derivation and portability are pretty crucial for wallet interoperability at this point, however that certainly wasn't the case when these BIPs were originally authored (I assume). Could they "graduate" in that sense?